src/screens/internal/Anomalies/AnomaliesSelector.tsx

```diff
@@ -0,0 +1,160 @@
+  );
+
+  const [stateFips, setLocations] = useState<string>(
+    getStateFipsParamValue(params, '01'),
+  );
+
+  useEffect(() => {
+    onNewOptions({
+      snapshot,
+      stateFips,
+      annotationType,
+    });
+  }, [snapshot, stateFips, annotationType, onNewOptions]);
+
+  function setQueryParams(newParams: {
+    snapshot?: number;
+    annotationType?: number;
+    stateFips?: string;
+  }) {
+    const params = {
+      snapshot: snapshot,
+      annotationType: annotationType,
+      stateFips: stateFips,
+      ...newParams,
+    };
+
+    history.push({
+      ...location,
+      search: QueryString.stringify(params),
+    });
+  }
+
+  const changeSnapshot = () => {
+    const snapshot = parseInt(snapshotText);
+    if (!Number.isNaN(snapshot)) {
+      setSnapshot(snapshot);
+      setQueryParams({ snapshot });
+    }
+  };
+
+  // TODO: Figure out correct type for event.
+  const changeMetric = (event: any) => {
+    const annotationType = parseInt(event.target.value);
+    setMetric(annotationType);
+    setQueryParams({ annotationType });
+  };
+
+  // TODO: Figure out correct type for event.
+  const changeLocation = (event: any) => {
+    const value = event.target.value;
+    const stateFips = String(value);
+    setLocations(stateFips);
+    setQueryParams({ stateFips });
+  };
+
+  return (
+    <OptionsSelectorWrapper>
+      <FormControl style={{ width: '8rem', marginRight: '1rem' }}>
+        <TextField
+          id="snapshot"
+          label="Snapshot"
+          value={snapshotText}
+          onChange={e => setSnapshotText(e.target.value)}
+          onBlur={() => changeSnapshot()}
+          onKeyPress={ev => {
+            if (ev.key === 'Enter') {
+              changeSnapshot();
+              ev.preventDefault();
+            }
+          }}
+        />
+      </FormControl>
+      <FormControl style={{ width: '12rem', marginLeft: '1rem' }}>
+        <InputLabel focused={false}>Field:</InputLabel>
+        <Select value={annotationType} onChange={changeMetric}>
+          {ANNOTATION_TYPES.map(annotationType => (
+            <MenuItem key={annotationType} value={annotationType}>
+              {annotationTypeNames[annotationType]}
+            </MenuItem>
+          ))}
+        </Select>
+      </FormControl>
+      <FormControl style={{ width: '14rem', marginLeft: '1rem' }}>
+        <InputLabel focused={false}>State:</InputLabel>
+        <Select value={stateFips} onChange={changeLocation}>
+          {regions.states.map(state => (
+            <MenuItem key={state.fipsCode} value={state.fipsCode}>
+              {state.name}
+            </MenuItem>
+          ))}
+        </Select>
+      </FormControl>
+    </OptionsSelectorWrapper>
+  );
+}
+
+function getStateFipsParamValue(
+  params: QueryString.ParsedQuery,
+  defaultValue: string,
+): string {
+  let value = get(params, 'stateFips', defaultValue);
+  if (typeof value !== 'string' || regions.findByStateCode(value)) {
```

Comment:
Seems like this should be `regions.findByStateCode(value) !== null` or something.

Alternatively we could probably just do a `regions.findByStateCodeStrict(value)` which will throw if it's not found.

Reply:
whoops, yeah I think you're right. I think I check `typeof value !== 'string'` first because `findByStateCode` only takes strings, so I couldn't pass `value` directly. But, in this format, I think it should be 

`if (typeof value !== 'string' || regions.findByStateCode(value)) === null`